bin/phraseapp_updater

```diff
@@ -16,16 +16,47 @@ class PhraseAppUpdaterCLI < Thor
   method_option :parent_commit,          type: :string, required: true, desc: 'git commit hash of initial locales'
   method_option :remove_orphans,         type: :boolean, default: true, desc: 'Remove keys not in the uploaded default locale'
 
+  # Options that mirror the PhraseApp API (https://developers.phrase.com/api/#post-/projects)
+  method_option :autotranslate_check_new_locales,                       type: :boolean, desc: 'Requires autotranslate_enabled to be true'
+  method_option :autotranslate_check_new_translation_keys,              type: :boolean, desc: 'Requires autotranslate_enabled to be true'
+  method_option :autotranslate_check_new_uploads,                       type: :boolean, desc: 'Requires autotranslate_enabled to be true'
+  method_option :autotranslate_enabled,                                 type: :boolean, desc: 'Autopilot, requires machine_translation_enabled.'
+  method_option :autotranslate_mark_as_unverified,                      type: :boolean, desc: 'Requires autotranslate_enabled to be true'
+  method_option :autotranslate_use_machine_translation,                 type: :boolean, desc: 'Requires autotranslate_enabled to be true'
+  method_option :autotranslate_use_translation_memory,                  type: :boolean, desc: 'Requires autotranslate_enabled to be true'
+  method_option :enable_all_data_type_translation_keys_for_translators, type: :boolean, desc: 'Otherwise, translators are not allowed to edit translations other than strings'
+  method_option :enable_icu_message_format,                             type: :boolean, desc: 'We can validate and highlight your ICU messages.'
+  method_option :machine_translation_enabled,                           type: :boolean, desc: 'Enable machine translation support in the project. Required for Pre-Translation'
+  method_option :point_of_contact,                                      type: :string,  desc: 'User ID of the point of contact for the project.'
+  method_option :workflow,                                              type: :string,  desc: 'Review Workflow. "simple" / "review".'
+  method_option :zero_plural_form_enabled,                              type: :boolean, desc: 'Displays the input fields for the \'ZERO\' plural form for every key as well although only some languages require the \'ZERO\' explicitly.'
```

Comment:
I had a couple of ideas for this originally.

The first idea was just to allow the use of `--` and anything beyond it would be passed directly to phraseapp.

```
phraseapp_updater setup \
  --phraseapp_project_name=... \
  --parent_commit=... \
  --phraseapp_api_key=... \
  --file-format=json \
  path/to/locales
  -- --zero_plural_form_enabled=false
```

But, I wasn't sure if `thor` would interpret the `false` here or if I would be able to convert those options to anything without significant magic sauce. So I went ahead and added my own definitions.

To try to make people's lives slightly less miserable, I added all the optional options that didn't pertain to branching.

Reply:
> only to add the options I actually wanted to change.

That'd be a pretty reasonable choice, to be honest. We use so little of Phraseapp that exposing too many of its options feels like it's more likely to cause rather than save trouble.

Reply:
I added a commit which does this. `method_options` actually returns a meta object with the information you provided. So I just capture them into a collection inline and use that later:

```ruby
PHRASEAPP_CREATE_PROJECT_OPTIONS = [
  method_option(:zero_plural_form_enabled, type: :boolean, desc: 'Displays the input fields for the \'ZERO\' plural form for every key as well although only some languages require the \'ZERO\' explicitly.')
]

def setup
  ...
  phraseapp_create_options = PHRASEAPP_CREATE_PROJECT_OPTIONS.map { |opt| opt.name.to_sym }
  phraseapp_opts = options.slice(*phraseapp_create_options)
  ...
end
```

Not sure if this is unorthodox or not though. I tested it and works as expected. Let me know if there's a way I can finesse this. Otherwise I think it's gtg.

Reply:
I'd say the code is a little surprising to read because the constant array looks like you're just preparing pure values, but actually each `method_option()` call is making a side-effect on the current class. It's entirely valid code, just unexpected.

I think it'd be less surprising to drive it along the lines of:

```ruby
PHRASEAPP_OPTIONS = {
 zero_plural_form_enabled: {
   type: :boolean,
   desc: 'Displays the input fields for the \'ZERO\' plural form for every key as well although only some languages require the \'ZERO\' explicitly.',
  },
}

PHRASEAPP_OPTIONS.each do |name, params|
  method_option(name, **params)
end

[...]

def setup
  [...]
  phraseapp_opts = options.slice(*PHRASEAPP_OPTIONS.keys)
  [...]
end
```

but that's a bit of a nitpick.

Reply:
> I'd say the code is a little surprising to read because the constant array looks like you're just preparing pure values, but actually each method_option() call is making a side-effect on the current class. It's entirely valid code, just unexpected.

> but that's a bit of a nitpick.

Nah, I agree with you rationale.

> I think it'd be less surprising to drive it along the lines of:

Thanks for the code!